src/FsAutoComplete.Core/Commands.fs

```diff
@@ -41,7 +41,7 @@ type CoreResponse =
     | FormattedDocumentationForSymbol of xmlSig: string * assembly: string * xmlDoc: string list * signature: (string * (string [] * string [] * string [] * string [] * string [] * string [])) * footer: string * cn: string
     | TypeSig of tip: FSharpToolTipText<string>
     | CompilerLocation of fcs: string option * fsi: string option * msbuild: string option
-    | Lint of file: string * warnings: LintWarning.Warning list
+    | Lint of file: string * warningsWithCodes: (string * string option * LintWarning.Warning) list
```

Comment:
can you explode the tuple so we can use named args? it help a lot maintenance to see what are that `string * string option`, more so if of same type, to not flip
or use a new type

Reply:
yeah I'll make this a proper type.  I was a bit clear on what the 'interface' of FSAC was, so I tried to keep the changes as minimal as possible.

Reply:
That type `CoreResponse` is internal only, is used to abstract the return of commmands so these can be serialized with the right format in stdio or LSP mode.
It's going away when we remove stdio

I think that DU with names are ok

```fsharp
    | CompilerLocation of fcs: string option * fsi: string option * msbuild: string option
```

but with tuples become hard soon

```fsharp
    | HelpText of name: string * tip: FSharpToolTipText * additionalEdit: (string * int * int * string) option
```